### PR TITLE
Add a way to update an app's desktop info

### PR DESCRIPTION
**Pull request by joaquimrocha**
These changes add a DBus method to make the desktop aware that something about an application's desktop info has changed and, currently, uses that trigger to re-read an app's desktop info and update its icon in the desktop.
The use case is a weblink that can be added to the desktop before its favicon has been retrieved; thus, once the favicon has been downloaded and the desktop file changed, the app icon will be updated o the desktop with these changes.

[endlessm/eos-shell#4920]

### CODE REVIEW

File: js/ui/appDisplay.js [hunk 1]
```diff
@@ -362,10 +362,20 @@ const EndlessApplicationView = new Lang.Class({
...
+                icon.icon.reloadIcon();
+            }
+        }
+    },
+
     addIcons: function(isHidden) {
         // Don't do anything if we don't have more up-to-date information, since
         // re-adding icons unnecessarily can cause UX problems
         if (!this.iconsNeedRedraw()) {
+            this._reloadIcons();
```
Comment (cosimoc) [inline js/ui/appDisplay.js]: Just a drive-by heads up that always doing I/O from this code path unfortunately **is not** acceptable. The `iconsNeedRedraw()` system was added exactly to avoid that. It's true that reloading the visible icons only is a little better than what we were doing back then (IIRC calling `g_app_info_get_all()`) but this is still a hot path, so the smarter solution may be really what we have to do here.

Comment (ebassi) [inline js/ui/appDisplay.js]: I have to agree with @cosimoc here: I/O in the compositor's main thread is pretty much going to cause interactivity issues on less powerful hardware platforms.

Comment (joaquimrocha) [inline js/ui/appDisplay.js]: Okay. That makes sense. I thought it'd not be that problematic, but it definitely doesn't scale so I will go for a smarter approach.

File: js/ui/appDisplay.js [hunk 2]
```diff
@@ -591,6 +593,13 @@ const AllView = new Lang.Class({
         }
      },
 
+     _onAppInfoChanged: function(app_system, app) {
+        for (let icon of this._allIcons) {
+            if (icon.app && icon.app === app)
+                icon.icon.reloadIcon();
```
Comment (ebassi) [inline js/ui/appDisplay.js]: Should probably break after `reloadIcon()`; we don't allow duplicates.

**Comment by joaquimrocha** [comment]
Updated after your comments @ebassi .
